src/kw/kw.cpp

```diff
@@ -146,23 +188,48 @@ string KanoWorld::get_refresh_header(string token)
 
 
 /**
- *   libcurl callback function to collect response from the server
- *
+ *   libcurl callback function, allows to collect the data response from the server,
+ *   which in this case is a json object.
  */
-size_t KanoWorld::write_function(void *ptr, size_t size, size_t nmemb, void *stream)
+size_t KanoWorld::write_function(void *ptr, size_t size, size_t nmemb, void *user_data)
 {
-    // TODO: Will be needed to parse the json response. How to collect class this pointer?
-    //printf(">>> RESPONSE: %s\n", ptr);
+    // Save the server's response in a class string
+    KanoWorld *pkw = (KanoWorld *) user_data;
+    pkw->server_response = string((char *) ptr);
+
     return size * nmemb;
 }
 
 
 /**
- *   Not implemented yet
+ *   Returns true when the cached token duration time has not expired yet,
+ *   false otherwise. Call the refresh_token API in such case.
  */
-bool KanoWorld::am_i_logged_in(void)
+bool KanoWorld::am_i_logged_in(bool verbose)
 {
-    return false;
+    std::time_t now = std::time(nullptr);
+
+    if (!load_data()) {
+        return false;
+    }
+
+    std::time_t duration = atol(expiration_date.c_str());
+    double seconds;
+
+    if (!duration) {
+        return false;
+    }
+
+    seconds = difftime(now, duration);
```

Comment:
Move the definition above to here:

```suggestion
    double seconds = difftime(now, duration);
```